x-pack/plugins/fleet/server/services/package_policy.ts

```diff
@@ -1887,33 +1906,32 @@ async function validateIsNotHostedPolicy(
 }
 
 export async function updatePackagePolicyVersion(
-  packagePolicy: (NewPackagePolicy & { version?: string; id: string }) | UpdatePackagePolicy,
   soClient: SavedObjectsClientContract,
+  packageName: string,
+  packageVersion: string,
   currentVersion?: string
 ) {
-  if (packagePolicy.package) {
-    await removeOldAssets({
-      soClient,
-      pkgName: packagePolicy.package.name,
-      currentVersion: packagePolicy.package.version,
-    });
+  await removeOldAssets({
+    soClient,
+    pkgName: packageName,
+    currentVersion: packageVersion,
+  });
 
-    if (packagePolicy.package.version !== currentVersion) {
-      const upgradeTelemetry: PackageUpdateEvent = {
-        packageName: packagePolicy.package.name,
-        currentVersion: currentVersion || 'unknown',
-        newVersion: packagePolicy.package.version,
-        status: 'success',
-        eventType: 'package-policy-upgrade' as UpdateEventType,
-      };
-      sendTelemetryEvents(
-        appContextService.getLogger(),
-        appContextService.getTelemetryEventsSender(),
-        upgradeTelemetry
-      );
-      appContextService.getLogger().info(`Package policy upgraded successfully`);
-      appContextService.getLogger().debug(JSON.stringify(upgradeTelemetry));
-    }
+  if (packageVersion !== currentVersion) {
+    const upgradeTelemetry: PackageUpdateEvent = {
+      packageName,
+      currentVersion: currentVersion || 'unknown',
+      newVersion: packageVersion,
+      status: 'success',
+      eventType: 'package-policy-upgrade' as UpdateEventType,
+    };
+    sendTelemetryEvents(
```

Comment:
This is changing a little the behavior here will not send one event per package policy but one for each package/version we may want to keep the telemetry the same as before, and move this to a separate method.